### ISSUE TITLE
Migrating data to distributed hypertable does not work

### ISSUE THREAD

**Issue by mkindahl** [timescale/timescaledb#1999]
**Relevant system information:**
 - OS: Ubuntu 19.10
 - PostgreSQL version (output of `postgres --version`): 12.2
 - TimescaleDB version (output of `\dx` in `psql`): 2.0.0-dev
 - Installation method: source

If a normal PostgreSQL table containing data is migrated to a distributed hypertable, it gives a strange error.

**To Reproduce**

```sql
Expanded display is used automatically.
Null display is "[NULL]".
psql (12.2)
Type "help" for help.

postgres=# SELECT * FROM add_data_node('dn1', host => 'localhost', port => 5441, if_not_exists => true);
NOTICE:  database "postgres" already exists on data node, skipping
 node_name |   host    | port | database | node_created | database_created | extension_created 
-----------+-----------+------+----------+--------------+------------------+-------------------
 dn1       | localhost | 5441 | postgres | t            | f                | t
(1 row)

postgres=# SELECT * FROM add_data_node('dn2', host => 'localhost', port => 5442, if_not_exists => true);
NOTICE:  database "postgres" already exists on data node, skipping
 node_name |   host    | port | database | node_created | database_created | extension_created 
-----------+-----------+------+----------+--------------+------------------+-------------------
 dn2       | localhost | 5442 | postgres | t            | f                | t
(1 row)

postgres=# SELECT * FROM add_data_node('dn3', host => 'localhost', port => 5443, if_not_exists => true);
NOTICE:  database "postgres" already exists on data node, skipping
 node_name |   host    | port | database | node_created | database_created | extension_created 
-----------+-----------+------+----------+--------------+------------------+-------------------
 dn3       | localhost | 5443 | postgres | t            | f                | t
(1 row)

postgres=# CREATE TABLE conditions(
postgres(#     time TIMESTAMPTZ NOT NULL,
postgres(#     device INTEGER,
postgres(#     temperature FLOAT,
postgres(#     humidity FLOAT
postgres(# );
CREATE TABLE
postgres=# 
postgres=# INSERT INTO conditions
postgres-# SELECT time, (random()*30)::int, random()*80 - 40, random()*100
postgres-# FROM generate_series('2019-01-01 00:00:00'::timestamptz, '2019-02-01 00:00:00', '1 min') AS time;
INSERT 0 44641
postgres=# SELECT * FROM create_distributed_hypertable('conditions', 'time', 'device', migrate_data => true);
NOTICE:  migrating data to chunks
DETAIL:  Migration might take a while depending on the amount of data.
ERROR:  no data nodes associated with hypertable "conditions"
```

**Expected behavior**

Data from original table is migrated to distributed hypertable.

**Actual behavior**

It gives a very strange error.

**Comment by mkindahl** [comment]
We block this initialy in #2230 and will fix it properly in a later release. Removing it from the 2.0.0 milestone for now.

**Comment by erimatnor** [comment]
Closing this issue since addressed in #2230. Let's open another issue if we'd like to support data migrations on distributed hypertables.
